src/Worker/Grpc/GrpcDurableTaskWorker.cs

```diff
@@ -48,7 +49,8 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
         await using AsyncDisposable disposable = this.GetCallInvoker(out CallInvoker callInvoker, out string address);
         this.logger.StartingTaskHubWorker(address);
-        await new Processor(this, new(callInvoker)).ExecuteAsync(stoppingToken);
+        IOrchestrationFilter? filter = this.services.GetService<IOrchestrationFilter>();
```

Comment:
Since `IOrchestrationFilter` is registered as a singleton, it will make more sense and be more efficient to resolve this as part of the constructor. As far as I can tell, the `GrpcDurableTaskWorker` is created via dependency-injection, so you can just add `IOrchestrationFilter` as an optional constructor parameter.

Reply:
Updated.